synapse/handlers/message.py

```diff
@@ -735,6 +697,126 @@ async def create_event(
 
         return event, context
 
+    async def create_event_for_batch(
```

Comment:
Can we call this from within `create_event`, rather than duplicating bits of the code? I see that we've split out some of the common functionality, which is great, but it might just be easier to call one from the other instead?